Assets/Scripts/EnemyAI.cs

```diff
@@ -195,17 +220,47 @@ private void OnDrawGizmos()
         Gizmos.color = Color.green;
         Gizmos.DrawWireSphere(position, chaseDistance);
         Gizmos.color = Color.red;
-        Gizmos.DrawWireSphere(position, attackDistance);
+        Gizmos.DrawWireSphere(position, maxAttackDistance);
     }
     
+    // TODO: enter a Grappled state
     public void OnGrappleStart()
     {
+        if (currentState == State.Dead) return;
         isGrappled = true;
+        animator.ResetTrigger("Flying");
+        animator.SetTrigger("Grappled");
+        animator.ResetTrigger("Shooting");
     }
 
+    // TODO: exit a Grappled state
     public void OnGrappleStop()
     {
+        if (currentState == State.Dead) return;
         isGrappled = false;
-        lastFireTime = Time.time;
+        if (distanceToPlayer < maxAttackDistance)
+        {
+            animator.SetTrigger("Shooting");
+            animator.ResetTrigger("Flying");
+            animator.ResetTrigger("Grappled");
+            lastFireTime = Time.time; // TODO: don't hardcode the animation transition state time
```

Comment:
make this `Time.time - 0.5f` to account for animation transition time (I think I had that in there at some point but it's somehow missing now)